src/main/kotlin/io/arrowkt/service/interpreter/AccountService.kt

```diff
@@ -0,0 +1,134 @@
+package io.arrowkt.service.interpreter
+
+import arrow.core.*
+import arrow.data.EitherT
+import arrow.data.Kleisli
+import arrow.effects.IO
+import io.arrowkt.Amount
+import io.arrowkt.ErrorOr
+import io.arrowkt.model.Account
+import io.arrowkt.model.Balance
+import io.arrowkt.repository.AccountRepository
+import io.arrowkt.service.*
+import io.arrowkt.service.AccountService
+import io.arrowkt.today
+import java.math.BigDecimal
+import java.time.LocalDate
+
+object AccountService : AccountService<Account, Amount, Balance> {
+
+    override fun open(
+        no: String,
+        name: String,
+        rate: Option<BigDecimal>,
+        openingDate: Option<LocalDate>,
+        accountType: AccountType
+    ): AccountOperation<Account> = Kleisli.invoke { repo ->
+        EitherT(
```

Comment:
The helper function, please :D Else it'll be too hard to read.

Reply:
the Tagless version is easier on the eye :wink: 

Type inference failed me for the helper function.... I couldn't figure it out...